utilities/transport.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of Go Responsiveness.
+ *
+ * Go Responsiveness is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 2 of the License, or (at your option) any later version.
+ * Go Responsiveness is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with Go Responsiveness. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package utilities
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"golang.org/x/net/http2"
+)
+
+func OverrideHostTransport(transport *http.Transport, connectToAddr string) error {
+	dialer := &net.Dialer{
+		Timeout: 10 * time.Second,
+	}
+
+	transport.DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+		_, port, err := net.SplitHostPort(addr)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(connectToAddr) > 0 {
+			addr = net.JoinHostPort(connectToAddr, port)
+		}
+
+		return dialer.DialContext(ctx, network, addr)
+	}
+
+	if err := http2.ConfigureTransport(transport); err != nil {
```

Comment:
Should we handle the case that this is not really an error:

> It returns an error if t1 has already been HTTP/2-enabled. 

That does not seem like a real "error" in this case.

Reply:
So just ignore it?

Reply:
It seems to me (according to the documentation) that the *only* error that could result is when HTTP/2 is already enabled. If you agree that is the case, then I would say, "yes, let's ignore it." I will defer to your judgment -- whatever way you decide is great.

Reply:
Removed. And actually this fixes up the error path for this where now we don't need to bubble the error up. 2 birds, 1 stone.

Reply:
Bingo! Thanks! 

I still think that we will want to have that "bubble-up path" but it's not as pressing now!